platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/event/tipset/TipsetEventCreatorImpl.java

```diff
@@ -268,10 +273,11 @@ private GossipEvent createEventByOptimizingAdvancementWeight() {
             }
         }
 
-        if (lastSelfEvent != null && bestOtherParent == null) {
-            // There exist no parents that can advance consensus, and this is not our first event.
-            // The only time it's ok to create an event with no other parent is when we are creating
-            // our first event.
+        if (bestOtherParent == null
+                && (minimumGenerationNonAncient > GENERATION_UNDEFINED + 1 || lastSelfEvent != null)) {
+            // If there are no available other parents, it is only legal to create a new event if we are
+            // creating a genesis event. In order to create a genesis event, we must have never created
+            // an event before and the current minimum generation non ancient must have never been advanced.
             return null;
         }
```

Comment:
It might just be me, but I found the logic as written here difficult to grasp - what do you think about this alternative layout? Up to you if you'd like it to stay as-is, though

```
        if (bestOtherParent == null) {
            // If there are no available other parents, it is only legal to create a new event if we are
            // creating a genesis event. In order to create a genesis event, we must have never created
            // an event before and the current minimum generation non ancient must have never been advanced.
            if (minimumGenerationNonAncient > GENERATION_UNDEFINED + 1 || lastSelfEvent != null) {
                // event creation isn't legal
                return null;
            }
            
            // we are creating a genesis event, so we can use a null other parent
            return buildAndProcessEvent(null);
        } else {
            tipsetMetrics.getTipsetParentMetric(bestOtherParent.getCreator()).cycle();
            return buildAndProcessEvent(bestOtherParent);
        }
```

Reply:
Refactored.

```
        if (bestOtherParent == null) {
            // If there are no available other parents, it is only legal to create a new event if we are
            // creating a genesis event. In order to create a genesis event, we must have never created
            // an event before and the current minimum generation non ancient must have never been advanced.
            if (minimumGenerationNonAncient > GENERATION_UNDEFINED + 1 || lastSelfEvent != null) {
                // event creation isn't legal
                return null;
            }

            // we are creating a genesis event, so we can use a null other parent
            return buildAndProcessEvent(null);
        }

        tipsetMetrics.getTipsetParentMetric(bestOtherParent.getCreator()).cycle();
        return buildAndProcessEvent(bestOtherParent);
```